### ISSUE TITLE
Pug - a template engine for nodejs and golang

### ISSUE THREAD

**Issue by coldume** [pugjs/pug#2343]
Hi,
I'm a golang developer, recently I found pug, and it is awesome. Although golang has its native template engine "html/template", it still uses ugly HTML syntax, lacks of template-level array definition, no built-in minify, etc. Pug fixes all of them.

Pug doesn't have to add new code to support golang, as it already does. Usually, I compiled the templates by pug first, then pass them to golang native template engine, letting it handles run-time markup like placeholders and run-time iterations.

Basically, I hope the description of  pug can change to "a template engine for nodejs and golang", and someone who has golang and pug background could write a tutorial for it, and add it to the pug website (I'm not native English speaker, so I cannot help too much). As golang becomes popular, It is great if we can officially invite the golang community to pug.

Thanks.

![pug_05_lg](https://cloud.githubusercontent.com/assets/5236124/14932323/574d8bce-0e42-11e6-9dd4-a06541b1e63c.jpg)

**Comment by TimothyGu** [comment]
Well Pug is written in JavaScript, and its API is only available to JavaScript consumers. Yes you can use it with other languages either by interaction through the CLI or by precompiling, but by this logic we would have to write "for Node.js, Go, PHP, C♯, Java, & the browser," which is a bit impractical.

**Comment by OscarRyz** [comment]
@coldume Do you have an example of how to use go and pug?

**Comment by shovon** [comment]
@oscarryz read what @TimothyGu said: pug has not been ported into Go, and there aren't any plans to ever import it into golang.

**Comment by oscarryz** [comment]
@shovon  Thanks for replaying. I was referring to this part: 

>_"Yes you can use it with other languages either by interaction **through the CLI**or by **precompiling**"_ 

I was looking for an example on how to use Go "through the CLI" or "precompiling"

**Comment by ForbesLindesay** [comment]
That would be a good question for stack overflow. You can find our CLI documentation at https://github.com/pugjs/pug-cli and you just need to find out how to invoke bash commands from go, which is a question for someone who knows about go, rather than for someone who knows about pug.
